packages/redux/src/products/reducer/__tests__/outfits.test.ts

```diff
@@ -0,0 +1,134 @@
+import { mockProductId } from 'tests/__fixtures__/products';
+import { productsActionTypes, ProductsOutfitsState } from '../..';
+import { toBlackoutError } from '@farfetch/blackout-client';
+import reducer, { getError, getIsLoading, INITIAL_STATE } from '../outfits';
+
+const mockAction = { type: 'foo' };
+const meta = { productId: mockProductId };
+let initialState: ProductsOutfitsState;
+
+describe('outfits redux reducer', () => {
+  beforeEach(() => {
+    initialState = reducer(INITIAL_STATE, mockAction);
+  });
+
+  describe('error() reducer', () => {
+    const error = 'An error occurred';
+    const expectedError = {
+      [mockProductId]: error,
+    };
+
+    it('should return the initial state', () => {
+      const state = reducer(INITIAL_STATE, mockAction).error;
+
+      expect(state).toEqual(initialState.error);
+    });
+
+    it('should handle FETCH_PRODUCT_OUTFITS_REQUEST action type', () => {
+      const expectedResult = { [mockProductId]: undefined };
+      const state = reducer(undefined, {
```

Comment:
You should test also for when the state contains an error for the same product id. To be fully correct, each slice reducer should be tested with 3 cases: 
  1. Reducer is called with the expected action and an undefined state;
  2. Reducer is called with the expected action and a state generated for the same producId by other action (like FETCH_PRODUCT_OUTFITS_FAILURE)
  3. Reducer is called with another action.

Right now, this test will test case 1. Can you at least test for case 2 on each test of this file? (If possible, the three tests would be great as well, but if it is a lot of work, at least 2 should be done as they are the most common scenario).

Reply:
Never mind case 3, it is already being done :)